server/service/mdm.go

```diff
@@ -1638,34 +1638,26 @@ func (svc *Service) BatchSetMDMProfiles(
 		return nil
 	}
 
+	var globalOrTeamID uint
+	if tmID != nil {
+		globalOrTeamID = *tmID
+	}
+
 	var profUpdates fleet.MDMProfilesUpdates
 	if profUpdates, err = svc.ds.BatchSetMDMProfiles(ctx, tmID, appleProfiles, windowsProfiles, appleDecls); err != nil {
 		return ctxerr.Wrap(ctx, err, "setting config profiles")
 	}
 
-	// set pending status for windows profiles
-	winProfUUIDs := []string{}
-	for _, p := range windowsProfiles {
-		winProfUUIDs = append(winProfUUIDs, p.ProfileUUID)
-	}
-	winUpdates, err := svc.ds.BulkSetPendingMDMHostProfiles(ctx, nil, nil, winProfUUIDs, nil)
+	// set pending status for host profiles
+	hostUpdates, err := svc.ds.BulkSetPendingMDMHostProfiles(ctx, nil, []uint{globalOrTeamID}, nil, nil)
```

Comment:
@mna @getvictor, this is the fix for the bug reported in #24322. The underlying issue there was that the gitops implementation calls this method at a point when the profile uuids are defaulted to empty strings (because we are generating the profile structs for the YAML paths rather than pulling from datastore records). Team/no team id is known at this point, so we can rely on that to do the bulk updates

Reply:
Oh I see, thanks for the clarification.